app/src/main/java/io/neurolab/main/MainActivity.java

```diff
@@ -35,19 +35,11 @@
 public class MainActivity extends AppCompatActivity
         implements NavigationView.OnNavigationItemSelectedListener, View.OnClickListener {
 
-    private int launcherSleepTime;
+    private static UsbManager usbManager;
+    private static UsbSerialDevice serialPort;
+    private static CommunicationHandler communicationHandler;
+    private static String deviceData;
     private final String ACTION_USB_PERMISSION = "io.neurolab.USB_PERMISSION";
-
-    private CardView focusButton;
-    private CardView relaxButton;
-    private CardView memGraphButton;
-    private Menu menu;
-
-    private UsbManager usbManager;
-    private UsbSerialDevice serialPort;
-    private CommunicationHandler communicationHandler;
-    private String deviceData;
-
     public UsbSerialInterface.UsbReadCallback readCallback = new UsbSerialInterface.UsbReadCallback() { //Defining a Callback which triggers whenever data is read.
         @Override
         public void onReceivedData(byte[] arg0) {
```

Comment:
@CloudyPadmal Currently, we are not using the data read in any other activity. Was testing in the TestModeActivity only first. But it seems the `devicedata` which is getting initailized here from the incoming `arg0` is `null`. Crosschecked this checked by Toasting the `deviceData`. Have provided the apk, you can have a look. Before connecting the Arduino board, remove the app from the background. Once the Arduino get's connected, it will automatically ask for perms for Neurolab to use the device and henceforth the app starts if user wants.

Reply:
If we are not getting data then we have a big issue with this implementation. A PR supposed to make a fix; not just some code that doesn't work :thinking:

Reply:
Yes @CloudyPadmal i agree. I tried things, but couldn't get the soul reason for why the data is recieved as null. Also logging in various section to track down the issue is also not possible for i need to disconnect from the adb to connect the arduino. Can you help me with the reason for the data not being received? Cause the Arduino serial port is getting detected correctly (i toasted the serial port as a string to check)

Reply:
Give me some time. I think I can fix the bug.